tripy/tests/frontend/module/test_sequential.py

```diff
@@ -153,6 +161,56 @@ def test_str_representation(self, dict_sequential_network):
         assert str(dict_sequential_network) == expected_str
 
 
+class TestMixedContainerSequential:
+    def test_basic_structure(self, mixed_container_sequential_network):
+        assert len(mixed_container_sequential_network) == 2
+        assert isinstance(mixed_container_sequential_network[0], tp.Conv)
+        assert callable(mixed_container_sequential_network[1])
+
+    def test_forward_pass(self, mixed_container_sequential_network):
+        input_data = tp.Tensor(tp.ones((1, 2, 2, 2), dtype=tp.float32))
+        output = mixed_container_sequential_network(input_data)
+        assert output.shape == [1, 2, 1, 1]
+
+    def test_named_children(self, mixed_container_sequential_network):
+        expected_names = [("0", mixed_container_sequential_network[0])]
+        assert list(mixed_container_sequential_network.named_children()) == expected_names
+
+    def test_state_dict(self, mixed_container_sequential_network):
+        state_dict = mixed_container_sequential_network.state_dict()
+        expected_keys = ["0.bias", "0.weight"]
+        assert list(state_dict.keys()) == expected_keys
+
+    def test_load_state_dict(self, mixed_container_sequential_network):
+        new_state_dict = {
+            "0.weight": tp.Parameter(tp.ones((2, 2, 1, 1), dtype=tp.float32)),
+            "0.bias": tp.Parameter(tp.zeros((2,), dtype=tp.float32)),
+        }
+        mixed_container_sequential_network.load_state_dict(new_state_dict, strict=False)
+
+        assert np.array_equal(
+            cp.from_dlpack(mixed_container_sequential_network[0].weight), cp.from_dlpack(new_state_dict["0.weight"])
+        )
+        assert np.array_equal(
+            cp.from_dlpack(mixed_container_sequential_network[0].bias), cp.from_dlpack(new_state_dict["0.bias"])
+        )
+
+    def test_str_representation(self, mixed_container_sequential_network):
+        # Check if the string representation includes both Conv and lambda avgpool
```

Comment:
Looks like the avgpool is being dropped here?

Reply:
yeah so that's a caveat. Because it is not a module, and it won't appear in `named_children`. If we wanted to show these callables, then we need to modify `__str__` for `Module` and also introduce some logic to keep track of callables. 

Option 2: name children can also include non modules but only use Module named_children in state dict?

Reply:
After discussing with @pranavm-nvidia we think modifying named_children is not a good idea. Since, state_dict won't include non-Module information anyways, it is fine for string representation not include it as well.

Hence, not changing anything.